src/resolve-dependency.ts

```diff
@@ -183,5 +200,5 @@ function resolvePackage (name, parent, job, cjsResolve) {
       return resolveFile(pathTarget, parent, job) || resolveDir(pathTarget, parent, job);
```

Comment:
This path should have `|| throwNotFound` on the fallback.

Reply:
Ah ha, this is the missing piece! I pushed a change which makes sure `resolveDependency(): string | string[]` so we never return undefined anymore ☺️